boot-qemu.py

```diff
@@ -733,8 +744,58 @@ def launch_qemu(cfg):
     gdb_bin = cfg["gdb_bin"]
     kernel_location = cfg["kernel_location"]
     qemu_cmd = cfg["qemu_cmd"]
+    share_folder_with_guest = cfg["share_folder_with_guest"]
     timeout = cfg["timeout"]
 
+    if share_folder_with_guest and not interactive:
+        utils.yellow(
+            'Shared folder requested without an interactive session, ignoring...'
+        )
+        share_folder_with_guest = False
+    if share_folder_with_guest and gdb:
+        utils.yellow(
+            'Shared folder requested during a debugging session, ignoring...')
+        share_folder_with_guest = False
+
+    if share_folder_with_guest:
+        shared_folder.mkdir(exist_ok=True, parents=True)
+
+        # If shared folder was requested, we need to search for virtiofsd in
+        # certain known locations.
+        qemu_prefix = Path(qemu_cmd[0]).resolve().parent.parent
+        virtiofsd_locations = [
+            Path('libexec', 'virtiofsd'),  # Default QEMU installation, Fedora
+            Path('lib', 'qemu', 'virtiofsd'),  # Arch Linux, Debian, Ubuntu
+        ]
+        virtiofsd = utils.find_first_file(qemu_prefix, virtiofsd_locations)
+
+        if not (sudo := shutil.which('sudo')):
+            raise Exception(
+                'sudo is required to use virtiofsd but it could not be found!')
+        utils.green(
+            'Requesting sudo permission to run virtiofsd in the background...')
+        subprocess.run([sudo, 'true'], check=True)
+
+        virtiofsd_log = base_folder.joinpath('.vfsd.log')
+        virtiofsd_mem = base_folder.joinpath('.vfsd.mem')
+        virtiofsd_socket = base_folder.joinpath('.vfsd.sock')
+        virtiofsd_cmd = [
+            sudo,
+            virtiofsd,
+            f"--socket-group={grp.getgrgid(os.getgid()).gr_name}",
+            f"--socket-path={virtiofsd_socket}",
+            '-o', f"source={shared_folder}",
+            '-o', 'cache=always',
+        ]  # yapf: disable
```

Comment:
This function is getting longer than I like.  Can some of this logic be split into separate functions called from here? (Can be a patch on top).

Reply:
For instance, the logic around `share_folder_with_guest` getting disabled, finding `virtiofsd`, testing `sudo`, and constructing `virtiofsd_cmd` seem like distinct operations that might make nice individual functions.

Reply:
Sure, I will add some helper functions here.

Reply:
2eca31e3da3c492cf71df4bfb3099eda7b056f10